### PR TITLE
Move documentation to Sphinx.

### LINKED ISSUE #6: Docs to RTD

### PR DESCRIPTION
**Pull request by jamadden**
Including having Sphinx run the doctests.

Fixes #6 

Docs are viewable at https://zopecontentprovider.readthedocs.io/en/docs

### CODE REVIEW

File: docs/motivation.rst [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+Before diving into the features of this package let me take up a few bytes of
+text to explain the use cases that drove us to develop this package (also
+others) and how the API documented below fulfills/solves those use cases. When
+we started developing Zope 3, it was from a desire to decentralize
+functionality and thus the complexity of the Python code. And we were
+successful! The component architecture is a marvelous piece of software that
+hopefully will allow us to build scalable solutions for a very long
+time. However, when it comes to user interface design, in this case
+specifically HTML pages, we have failed to provide the features and patterns
+of assembeling a page from configured components.
```
Comment (mgedmin) [inline docs/motivation.rst]: assembeling -> assembling

File: docs/tales.rst [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  ...                               provides=interfaces.IContentProvider,
+  ...                               name='mypage.MessageBox')
+
+The content provider must be registered by name, since the TALES expression
+uses the name to look up the provider at run time.
+
+Let's now create a view using a page template:
+
+  >>> import os, tempfile
+  >>> temp_dir = tempfile.mkdtemp()
```
Comment (mgedmin) [inline docs/tales.rst]: I would suggest `tempfile.mkdtemp(prefix='test-zopecontentprovider-')`

File: docs/tales.rst [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+  ...       <tal:block replace="structure provider:mypage.MessageBox" />
+  ...     </div>
+  ...     <div class="main">
+  ...       Content here
+  ...     </div>
+  ...   </body>
+  ... </html>
+  ... ''')
+
+As you can see, we exprect the ``provider`` expression to simply look up the
```
Comment (mgedmin) [inline docs/tales.rst]: exprect -> expect

File: docs/tales.rst [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+The event holds the provider and the request.
+
+  >>> events[0].request
+  <zope.publisher.browser.TestRequest instance URL=http://127.0.0.1>
+  >>> events[0].object
+  <MessageBox object at ...>
+
+Failure to lookup a Content Provider
+====================================
```
Comment (mgedmin) [inline docs/tales.rst]: lookup -> look up

File: docs/tales.rst [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+
+We'll change our message box content provider implementation a bit, so the new
+information is used:
+
+  >>> @zope.interface.implementer(IMessageType)
+  ... class BetterDynamicMessageBox(DynamicMessageBox):
+  ...     type = None
+  ...
+  ...     def render(self):
+  ...         return u'<div class="box,%s">%s</div>' %(self.type, self.message)
```
Comment (mgedmin) [inline docs/tales.rst]: Space after `%`

File: docs/tales.rst [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+:class:`zope.location.interfaces.ILocation`, then it will have its
+``__name__`` set to the name that was used to invoke it.
+
+
+  >>> from zope.location.interfaces import ILocation
+  >>> @zope.interface.implementer(ILocation)
+  ... class LocationDynamicMessageBox(BetterDynamicMessageBox):
+  ...
+  ...     def render(self):
+  ...         return u'<div class="box">%s</div>' %(self.__name__,)
```
Comment (mgedmin) [inline docs/tales.rst]: Space after `%`

File: src/zope/contentprovider/interfaces.py [hunk 7]
```diff
@@ -29,13 +29,15 @@ class IUpdateNotCalled(zope.interface.common.interfaces.IRuntimeError):
...
+    Default implementation of `IUpdateNotCalled`.
+    """
     def __init__(self, *args):
         if not args:
             args = ('``update()`` was not called yet.',)
         super(UpdateNotCalled, self).__init__(*args)
 
 class IBeforeUpdateEvent(IObjectEvent):
-
     """A Contentprovider will be updated"""
```
Comment (mgedmin) [inline src/zope/contentprovider/interfaces.py]: Contentprovider -> content provider

File: docs/tales.rst [hunk 8]
```diff
@@ -0,0 +1,322 @@
...
+
+  >>> @zope.interface.implementer(zope.interface.Interface)
+  ... class Content(object):
+  ...     pass
+
+  >>> content = Content()
+
+Finally we look up the view and render it. Note that a
+`.BeforeUpdateEvent` is fired - this event should always be fired before
+any contentprovider is updated.
```
Comment (mgedmin) [inline docs/tales.rst]: contentprovider -> content provider

**Review by mgedmin — APPROVED** [review]
LGTM, with minor fixes.

It's amazing how moving old text from one file to another can make reviewers discover typos in the old text ;)

**Comment by jamadden** [comment]
Thanks! 

> It's amazing how moving old text from one file to another can make reviewers discover typos in the old text ;)

I'm not sure "amazing" is the word I would use for it 😜

**Pull request merged**